editor/src/clj/editor/gui.clj

```diff
@@ -3351,34 +3303,59 @@
                                                 {:width w :height h}))))
   (input id-prefix g/Str)
   (output id-prefix g/Str (gu/passthrough id-prefix))
-  (output unused-display-profiles g/Any (g/fnk [own-layout-names display-profiles]
+  (output unused-display-profiles g/Any (g/fnk [layout-names display-profiles]
                                           (coll/transfer display-profiles []
                                             (map :name)
-                                            (remove own-layout-names)))))
+                                            (remove layout-names))))
+
+  (input aux-basic-gui-scene-info BasicGuiSceneInfo)
+  (output own-basic-gui-scene-info BasicGuiSceneInfo :cached
+          (g/fnk [font-names layer->index layer-names layout-names material-infos particlefx-resource-names spine-scene-names texture-names texture-resource-names spine-scene-element-ids]
+            {:font-names font-names
+             :layer->index layer->index
+             :layer-names layer-names
+             :layout-names layout-names
+             :material-infos material-infos
+             :particlefx-resource-names particlefx-resource-names
+             :spine-scene-element-ids (reduce coll/merge spine-scene-element-ids)
+             :spine-scene-names spine-scene-names
+             :texture-names texture-names
+             :texture-resource-names texture-resource-names}))
+  (output basic-gui-scene-info BasicGuiSceneInfo :cached
+          (g/fnk [aux-basic-gui-scene-info own-basic-gui-scene-info]
+            ;; Note: The layer configurations of any template sources are
+            ;; covered up by the layer configuration of the referencing scene.
+            (coll/merge-entries-with
+              (fn [[key :as aux-entry] own-entry]
+                (case key
+                  (:layer->index :layer-names) aux-entry ; Covered, not merged.
+                  (let [aux-value (val aux-entry)
+                        own-value (val own-entry)
+                        value (coll/merge aux-value own-value)]
+                    (pair key value))))
+              aux-basic-gui-scene-info
+              own-basic-gui-scene-info)))
```

Comment:
The special layer handing was the source of the template layer ordering issue when viewed in the Scene View. We were correctly covering the `layer-names` of the referenced scene by the ones from the referencing scene, but not the `layer->index` map.

Reply:
I have a suggestion: define `coll/merge-with-kv` and use it instead of `coll/merge-entries-with`. Make the merge fn accept 3 arguments: `k`, `old-v`, `new-v`, and return a merged `v`. This will make the code more efficient since we won't be allocating any entries.

Reply:
Good suggestion! I'll change it. It will make it cleaner at the call site as well.